tests/Providers/FindologicServiceProviderTest.php

```diff
@@ -0,0 +1,129 @@
+        $this->request = $this->getMockBuilder(Request::class)->disableOriginalConstructor()->setMethods([])->getMock();
+        $this->eventDispatcher = $this->getMockBuilder(Dispatcher::class)->disableOriginalConstructor()->setMethods([])->getMock();
+        $this->configRepository = $this->getMockBuilder(ConfigRepository::class)->disableOriginalConstructor()->setMethods([])->getMock();
+    }
+
+    public function testBootShopKeyNotSet()
+    {
+        $this->configRepository->expects($this->once())->method('get')->willReturn('');
+
+        $this->eventDispatcher->expects($this->never())->method('listen');
+
+        $this->runBoot();
+    }
+
+    public function testBootShopKeySetAndAliveTestFails()
+    {
+        $this->configRepository->expects($this->once())->method('get')->willReturn('testShopKey');
+
+        $this->eventDispatcher->expects($this->never())->method('listen');
+
+        $this->runBoot();
+    }
+
+    public function testIsNotSearchPageAndIsNotActiveOnCatPage()
+    {
+        $this->configRepository->expects($this->any())->method('get')->willReturnMap([
+            [
+                Plugin::CONFIG_SHOPKEY, false, 'testShopKey'
+            ]
+        ]);
+
+        $this->searchService->expects($this->once())->method('isAliveTest')->willReturn(true);
+
+        $this->eventDispatcher->expects($this->at(0))->method('listen')->with('IO.Resources.Import');
+        $this->eventDispatcher->expects($this->at(1))->method('listen')->with('Ceres.Search.Query');
+
+        $this->request->method('getUri')->willReturn('https://testshop.com/testpage');
+
+        $this->runBoot();
+    }
+
+    public function testIsSearchPageAndIsNotActiveOnCatPage()
+    {
+        $this->configRepository->expects($this->any())->method('get')->willReturnMap([
+            [
+                Plugin::CONFIG_SHOPKEY, false, 'testShopKey'
+            ]
+        ]);
+
+        $this->searchService->expects($this->once())->method('isAliveTest')->willReturn(true);
+
+        $this->eventDispatcher->expects($this->at(0))->method('listen')->with('IO.Resources.Import');
+        $this->eventDispatcher->expects($this->at(1))->method('listen')->with('Ceres.Search.Options');
+        $this->eventDispatcher->expects($this->at(2))->method('listen')->with('IO.Component.Import');
+        $this->eventDispatcher->expects($this->at(3))->method('listen')->with('Ceres.Search.Query');
+
+        $this->request->method('getUri')->willReturn('https://testshop.com/search');
+
+        $this->runBoot();
+    }
+
+    public function testIsNotSearchPageAndIsActiveOnCatPage()
+    {
+        $this->configRepository->expects($this->any())->method('get')->willReturn('testConfigValue');
+
+        $this->searchService->expects($this->once())->method('isAliveTest')->willReturn(true);
+
+        $this->eventDispatcher->expects($this->at(0))->method('listen')->with('IO.Resources.Import');
```

Comment:
Try to use `withConsecutive` instead of providing a call index, as this can give us a hard time in the future.